herddb-core/src/main/java/herddb/data/consistency/TableDataChecksum.java

```diff
@@ -0,0 +1,113 @@
+import herddb.core.DBManager;
+import herddb.model.Column;
+import herddb.model.Table;
+import herddb.sql.TranslatedQuery;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import net.jpountz.xxhash.StreamingXXHash64;
+import net.jpountz.xxhash.XXHashFactory;
+
+/**
+ * digest creation by scanning the table
+ *
+ * @author Hamado.Dene
+ */
+public abstract class TableDataChecksum {
+
+    private static final Logger LOGGER = Logger.getLogger(TableDataChecksum.class.getName());
+    private static final XXHashFactory factory = XXHashFactory.fastestInstance();
+    //private static final int SEED = 0x9747b28c;
+    private static final int SEED = 0;
+    public static final boolean DIGEST_NOT_AVAILABLE = true;
+    public static final String HASH_TYPE = "StreamingXXHash64";
+    public static int NRECORDS = 0;
+    public static long SCAN_DURATION = 0;
+    private static TranslatedQuery translated;
+
+    public static TableChecksum createChecksum(DBManager manager, TranslatedQuery query, TableSpaceManager tableSpaceManager, String tableSpace, String tableName) throws DataScannerException {
+
+        AbstractTableManager tablemanager = tableSpaceManager.getTableManager(tableName);
+        String nodeID = tableSpaceManager.getDbmanager().getNodeId();
+        if (query == null) {
+            final Table table = tableSpaceManager.getTableManager(tableName).getTable();
+            String columns = parseColumns(table);
+            translated = manager.getPlanner().translate(tableSpace, "SELECT  "
+                    + columns
+                    + " FROM " + tableName
+                    + " order by "
+                    + parsePrimaryKeys(table), Collections.emptyList(), true, false, false, -1);
+        }
+        translated = query;
+        ScanStatement statement = translated.plan.mainStatement.unwrap(ScanStatement.class);
+        LOGGER.log(Level.INFO, "creating checksum for table {0}.{1} on node {2}", new Object[]{tableSpace, tableName, nodeID});
+        manager.setAllowExecutionFromFollower(true);
+        try ( DataScanner scan = manager.scan(statement, translated.context, TransactionContext.NO_TRANSACTION);) {
+            StreamingXXHash64 hash64 = factory.newStreamingHash64(SEED);
+            byte[] serialize;
+            long _start = System.currentTimeMillis();
+            NRECORDS = 0; 
+            while (scan.hasNext()) {
+                NRECORDS++;
+                DataAccessor data = scan.next();
+                Object[] obj = data.getValues();
+                System.out.println("record numero = " + NRECORDS  + "valore = " + Arrays.toString(data.getValues()));
+                Column[] schema = scan.getSchema();
+                for (int i = 0; i < schema.length; i++) {
+                    serialize = RecordSerializer.serialize(obj[i], schema[i].type);
+                    hash64.update(serialize, 0, SEED);
```

Comment:
why do you need to pass this "SEED" ?

Reply:
You're right, i don't need to pass SEED here.
Update function need (byte [] buff, int offset, int len) were len is the of bytes to hash